### DISCUSSION TITLE
ある聖盃おまとめプルリクのコンフリクトについて話が…

### DISCUSSION THREAD

**Discussion by ShanShan1129** [orga-itsuka-trpg/TRPG-OCL-Rule#224] [General]
[聖盃おまとめ](https://github.com/orga-itsuka-trpg/TRPG-OCL-Rule/pull/222)で重複・競合が発生してるみたいだな　ルフィ  
これの解決についてはREADMEにも書いてないので、調べてもよくわからなかった場合はここで質問してくれ(^^)

**Comment by Mikabuki** [comment]
競合が発生している理由は恐らく練習卓が更新された分競合してるんだとは思うんすがこれはもうプルリクのタイミング的に仕方なかったッ
pdfとpngがある以上個別にログは(たぶん)投げられないからな...ルフィ
取り敢えず聖盃はここ数日は立たないと思うんでShanShan側のろぐが反映された今改めてこっちでForkして反映すれば問題はなさそうだとは思ってんすがね...
なんか変な所あったら教えてくれ(^^)

**Reply by ShanShan1129** [reply]
競合についてはルルブも含めておれが先に投げちゃったのでそれはそう
ただし一番最新のやつ(アリスとフランキーのやつ)はやってくれたのを確認したので、おれはログ投げてねェな…
なのでそこはお願いしてもいいすかね…？

そして今回のコンフリクトはMikabukiのmasterブランチが最新になってなかった🌈
のも理由の一つなので
まずは[これ](https://github.com/orga-itsuka-trpg/TRPG-OCL-Rule#2%E5%9B%9E%E7%9B%AE%E4%BB%A5%E9%99%8D%E3%81%AE%E3%83%97%E3%83%AB%E3%83%AA%E3%82%AF%E5%8D%94%E5%8A%9B%E3%82%B7%E3%83%A3%E3%83%B3%E5%90%91%E3%81%91)を参考に最新に更新しておくことを…勧める…
あとプルリクする際はforkした後、更にbranchを作ってそれにファイル追加するようにすると事故が発生してもブランチ消去で捲き戻しできるのでおすすめ

**Reply by ShanShan1129** [reply]
追伸：もしかするとファイルの関係で最新の状態にできない可能性もあるのでその時はまた教えてくれ(^^)

**Comment by Mikabuki** [comment]
かと思ったらpdfの名称が競合起こしてるだけで他ファイルは大丈夫っぽいんすがね？
それならpdfの送信をとりやめるだけでどうにか...できればいいんすがね...
Forkしたのは今回が初めてだったはずなんだがなんで最新になってなかったのかは...←意味深に無言
branchの存在は後で気づいたので藤虎です　次回以降参考にするすまん(thanks)
最新取得がプルリク送ってるせいかできない状態(解消を求められる)なので一度プルリクを取りやめて更新する(?)か競合ファイルだけ修正してプルリク送りなおすか...って感じがいいんすがね？

**Reply by Mikabuki** [reply]
リプライ見逃してた🌈
おれ英語嫌いだ　吐き気がする

**Reply by Mikabuki** [reply]
いや違うか　どちらにせよ最新版は更新しないとどうしようもねェんだな
取り敢えずファイル名再度設定して送りなおすぞ　来い

**Reply by ShanShan1129** [reply]
悲しいことにgithub単体だとファイル送信の取り消しはできないことをｵｯｼｪﾙ
そしてどうも活動ログ的に6月の頭にforkを一度やってたみたいだな　ルフィ
でも始めたばっかの時は、どのボタンがどう作用するとか全くわからないはずなので仕方がなかったっ　おれもわかんなかったし

ただ現状は改善しなければならないので、やってもらいたいこととして…あるfork消去について話が…
今のところおれも力が至らず、いまのMikabukiがforkしている`TRPG-OCL-Rule`はファイル関係がややこしいことになっている…助からぬ！なので
Mikabukiのforkした`TRPG-OCL-Rule`を削除して新しくforkし直すしか解決策が思いつかない　すまん(Thanks.)
[ここ](https://qiita.com/tmiame/items/f5dae82765ecf3d98889)を参考にMikabukiのリポジトリからforkしたものを削除してくれ(^^)

そしてMikabukiの作業がすべて消えてしまうことになるのは申し訳ないが、もう一度改めて作業をお願いしても…だめ゛かな゛ー！！！

**Reply by Mikabuki** [reply]
とりあえず作業内容差戻＋再アップロードは並行してやってるのでそれはそれで大丈夫だろ
なんかすごく手間かけて申し訳ないな

最新状態の反映をリクエスト履歴から確保→別途ブランチを作成

でどうやらForkを消さないでも最新版は実質確保できそうだな...ルフィ
取り敢えず作業は進める　これでダメだったらＦｏｒｋ削除してやり直しだな、ルフィ

**Reply by ShanShan1129** [reply]
お前なら…いい
いろいろやってくれるのはありがてェので必要な技術提供は惜しまないぞ　来い
それに
>最新状態の反映をリクエスト履歴から確保→別途ブランチを作成
>でどうやらForkを消さないでも最新版は実質確保できそうだな...ルフィ

しらそん…
とまあおれも知らないことまだまだたくさんなので、おれがしくじったときは助けてくれ(^^)

**Reply by Mikabuki** [reply]
わかった
助けてくれてうわあり！

**Reply by ShanShan1129** [reply]
追伸：
プルリクエスト毎にブランチを分けると取り回しやすいってチョッパーが言ってたぞ
あとはforkをするときはブランチ元がmasterであるか確認することを…勧める…(1敗)
